### ISSUE TITLE
v3.5: Wrong TS types for `useLazyQuery`

### ISSUE THREAD

**Issue by aaronadamsCA** [apollographql/apollo-client#9116]
**Intended outcome:**
Successfully upgrade Apollo Client from v3.4 to v3.5.

**Actual outcome:**
TypeScript ESLint warning on this code:

```tsx
  useEffect(() => {
    if (!called && inView) {
      void runQuery();
    }
  }, [called, inView, runQuery]);
```

The `!called` check is getting `Unnecessary conditional, value is always falsy`. That's because according to the types, `called` is always `true`, even for a lazy query. That's obviously incorrect.

There was no such type mismatch in v3.4.

**How to reproduce the issue:**
Check `called` in any lazy query in a TypeScript codebase. You'll see it permanently equals `true`, according to the types:

<img src="https://user-images.githubusercontent.com/1505561/143294045-93d4c4bd-2317-400c-9dce-28ce29ed9095.png" width="250" />

I see multiple problems in the existing types:

- `useQuery` shouldn't return `called` at all, [according to the docs:](https://www.apollographql.com/docs/react/api/react/hooks/#called)

  > This field is only present on the result object returned by `useLazyQuery`.

- Per the above note, then, the result object returned _when calling_ the lazy query shouldn't include `called`, either.

- The result object returned by `useLazyQuery` should include `called: boolean`.

**Versions**

```
  System:
    OS: Linux 4.19 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
  Binaries:
    Node: 16.13.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/bin/yarn
    npm: 8.1.0 - /usr/local/bin/npm
```

**Comment by aaronadamsCA** [comment]
Patched locally with:

```patch
diff --git a/node_modules/@apollo/client/react/types/types.d.ts b/node_modules/@apollo/client/react/types/types.d.ts
index 845cbb0..4ce1bb0 100644
--- a/node_modules/@apollo/client/react/types/types.d.ts
+++ b/node_modules/@apollo/client/react/types/types.d.ts
@@ -32,7 +32,6 @@ export interface QueryResult<TData = any, TVariables = OperationVariables> exten
     error?: ApolloError;
     loading: boolean;
     networkStatus: NetworkStatus;
-    called: true;
 }
 export interface QueryDataOptions<TData = any, TVariables = OperationVariables> extends QueryFunctionOptions<TData, TVariables> {
     children?: (result: QueryResult<TData, TVariables>) => ReactNode;
@@ -48,9 +47,11 @@ export interface QueryLazyOptions<TVariables> {
     variables?: TVariables;
     context?: DefaultContext;
 }
-export declare type LazyQueryResult<TData, TVariables> = QueryResult<TData, TVariables>;
+export interface LazyQueryResult<TData, TVariables> extends QueryResult<TData, TVariables> {
+    called: boolean;
+};
 export declare type QueryTuple<TData, TVariables> = [
-    (options?: QueryLazyOptions<TVariables>) => Promise<LazyQueryResult<TData, TVariables>>,
+    (options?: QueryLazyOptions<TVariables>) => Promise<QueryResult<TData, TVariables>>,
     LazyQueryResult<TData, TVariables>
 ];
 export declare type RefetchQueriesFunction = (...args: any[]) => InternalRefetchQueriesInclude;
```

I'd be happy to submit a PR, but I have no idea how to write a test for this.

**Comment by aaronadamsCA** [comment]
Hey @benmosher, I see you did #5935 a while back, which (indirectly?) led to this. So you might be the best person to comment on whether the fix I've suggested would be the right one.

**Comment by anark** [comment]
Same as https://github.com/apollographql/apollo-client/issues/9102.  Would love to see a fix for this.

**Comment by aaronadamsCA** [comment]
Oops, yes, this is definitely a duplicate of #9102.

But I'm going to passive-aggressively leave it to a maintainer to close this, so we can actually get acknowledgment of the issue first :)

**Comment by benjamn** [comment]
The typing of `called` should be fixed in v3.5.8, thanks to #9328.
